src/libs/blueprint/conduit_blueprint_mesh.cpp

```diff
@@ -1985,6 +1985,94 @@ mesh::number_of_domains(const conduit::Node &mesh)
     }
 }
 
+//-------------------------------------------------------------------------
+void
+mesh::state(const conduit::Node &mesh, conduit::Node &state)
+{
+    state.reset();
+    if (! is_multi_domain(mesh))
+    {
+        if (mesh.has_child("state"))
+        {
+            state.set(mesh["state"]);
+        }
+    }
+    else
+    {
+        auto doms_itr = mesh.children();
+        while (doms_itr.has_next())
+        {
+            const Node &dom = doms_itr.next();
+            if (dom.has_child("state"))
+            {
+                state.set(dom["state"]);
+                break;
+            }
+        }
+    }
+}
+
+//-------------------------------------------------------------------------
+index_t
+mesh::cycle(const conduit::Node &mesh)
+{
+    Node state;
+    ::conduit::blueprint::mesh::state(mesh, state);
+    if (state.has_child("cycle"))
+    {
+        return state["cycle"].to_index_t();
+    }
+    return std::numeric_limits<int>::max();
+}
+
+//-------------------------------------------------------------------------
+index_t
+mesh::cycle(const conduit::Node &mesh, 
+            const std::string &path,
+            bool just_asking)
+{
+    Node state;
+    ::conduit::blueprint::mesh::state(mesh, state);
+    if (state.has_child("cycle"))
+    {
+        return state["cycle"].to_index_t();
+    }
+    CONDUIT_INFO("is_cycle_present: no 'state/cycle' present."
+                 " Defaulting to counter");
+    static std::map<std::string, int> counters;
+    index_t cycle = counters[path];
+    // if we're just using this function to ask what the cycle could be, 
+    // then we don't want to mess with the static counter. If we are
+    // in the process of writing files, then we can make up a new cycle.
+    if (! just_asking)
+    {
+        counters[path] ++;
+    }
+    return cycle;
+}
+
+//-------------------------------------------------------------------------
+bool
+mesh::is_cycle_present(const conduit::Node &mesh)
+{
+    Node state;
+    ::conduit::blueprint::mesh::state(mesh, state);
+    return state.has_child("cycle");
+}
```

Comment:
I had to separate `is_cycle_present()` from cycle calculations because unifying them causes things to fall apart. Namely, if we generate a cycle for data with no cycle but we didn't end up needing to (like in the case where the requested suffix is none or default), then we have incremented our counter when we didn't have to, breaking several tests, just because we wanted to know if the cycle was present or not. Separating it into two functions avoids this issue.

Reply:
`is_cycle_present()` is gone now